src/disk_io_thread.cpp

```diff
@@ -193,63 +197,25 @@ namespace libtorrent
 	void disk_io_thread::abort(bool wait)
 	{
 		m_abort = true;
-		if (m_num_threads == 0)
+		if (num_threads() == 0)
 		{
 			abort_jobs();
 		}
 		else
 		{
-			set_num_threads(0, wait);
+			m_generic_threads.abort(wait);
```

Comment:
are you sure we don't need to call abort_jobs() here too? to cancel the ones that haven't started yet?